libs/base/Language/Reflection/TTImp.idr

```diff
@@ -209,22 +248,34 @@ mutual
                    (opts : List FnOpt) ->
                    (type : ITy) ->
                    IClaimData
-
+  ||| A top-level declaration
   public export
   data Decl : Type where
-       IClaim : WithFC IClaimData -> Decl
-       IData : FC -> WithDefault Visibility Private -> Maybe TotalReq -> Data -> Decl
-       IDef : FC -> Name -> (cls : List Clause) -> Decl
+       ||| A type ascription, `a : b`.
+       ||| Called a claim because of Curry Howard, the statement `x : p` is equivalent to `x` is a proof of `p`.
+       IClaim : (claim : WithFC IClaimData) -> Decl
+       ||| A data declaration
+       IData : FC -> (vis : WithDefault Visibility Private) -> Maybe TotalReq -> (cons : Data) -> Decl
+       ||| A function definition by its clauses
+       IDef : FC -> (f : Name) -> (cls : List Clause) -> Decl
+       ||| A parameter block
        IParameters : FC -> (params : List (Name, Count, PiInfo TTImp, TTImp)) ->
                      (decls : List Decl) -> Decl
+       ||| A record declaration
+       ||| @ ns Nested namespace
        IRecord : FC ->
-                 Maybe String -> -- nested namespace
-                 WithDefault Visibility Private ->
-                 Maybe TotalReq -> Record -> Decl
+                 (ns : Maybe String) ->
+                 (vis : WithDefault Visibility Private) ->
+                 (totality : Maybe TotalReq) -> (rec : Record) -> Decl
+       ||| A namespace declaration, `namespace ns where decls`
```

Comment:
In namespace syntax where is not `where`, a newline with a indent or `{ ... }` should go there